### PR TITLE
[capnproto] Update to 0.9.1

### PR DESCRIPTION
**Pull request by chausner**
**Describe the pull request**

- #### What does your PR fix?  
  Updates capnproto to 0.9.1

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  As before, Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run ./vcpkg x-add-version --all and committed the result?  
  Yes

### CODE REVIEW

File: ports/capnproto/portfile.cmake [hunk 1]
```diff
@@ -8,26 +8,33 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
...
+    SHA512 BDC5960F26214139744D768CA4558B085E2D359C375A8849CA532A4C458A2DAF0EE1A1F4D00745DE5ABF353AA1E4084A2D85E5523006F23F591F21FAC2D847FD
     HEAD_REF master
 )
 
-vcpkg_configure_cmake(
-    SOURCE_PATH ${SOURCE_PATH}
-    PREFER_NINJA
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+  FEATURES
+    "openssl" OPENSSL_FEATURE
```
Comment (NancyLi1013) [inline ports/capnproto/portfile.cmake:18]:
```suggestion
vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
    FEATURES
        "openssl" OPENSSL_FEATURE
```
There should be four spaces.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Hi @chausner
Thanks for your PR.
Have you tested the feature openssl?

**Comment by ssrlive** [comment]
Current version told me this error
```
error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' 
```
please check it again.

**Comment by chausner** [comment]
> Have you tested the feature openssl?

I have tested that it compiles without issues on x64-windows and that it links to OpenSSL as expected. I have not tested whether the library functions correctly.

File: ports/capnproto/portfile.cmake (hunk 1 above)
Comment (chausner) [reply ports/capnproto/portfile.cmake]: Fixed in e89120bcf7e16f01c08c14762ba6661bfad67a0f.

**Comment by NancyLi1013** [comment]
> Current version told me this error
> 
> ```
> error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' 
> ```
> 
> please check it again in your new version.

Hi @ssrlive 

Could you please help confirm how did you encounter this problem? If possible, please help provide the steps to reproduce.

Thanks.

**Comment by NancyLi1013** [comment]
> > Hi @ssrlive
> > Could you please help confirm how did you encounter this problem? If possible, please help provide the steps to reproduce.
> 
> I think @ssrlive refers to an issue with capnproto 0.8.0, correct?

@ssrlive Please help confirm on which version you encountered the issue. Thanks.

**Review by NancyLi1013 — APPROVED** [review]

**Pull request merged**